src/frontend/src/binder/relation/table_or_source.rs

```diff
@@ -154,7 +155,8 @@ impl Binder {
                     let user_name = &self.auth_context.user_name;
 
                     for path in self.search_path.path() {
-                        if path == PG_CATALOG_SCHEMA_NAME {
+                        if path == PG_CATALOG_SCHEMA_NAME || path == PG_OPCLASS_TABLE_NAME {
```

Comment:
This's not necessary. The search path is only about the schema namespace, not the table name.